src/Symfony/Component/Lock/Store/DoctrineDbalStore.php

```diff
@@ -235,4 +246,21 @@ private function getCurrentTimestampStatement(): string
                 return (string) time();
         }
     }
+
+    /**
+     * Checks wether current driver supports table creating within transaction.
+     */
+    private function driverSupportsTableCreationInTransaction(): bool
+    {
+        $driver = $this->conn->getDriver();
+
+        switch (true) {
+            case $driver instanceof AbstractPostgreSQLDriver:
```

Comment:
If I remember well, theses classes have been renamed in Doctrine 3 
see `getCurrentTimestampStatement` method for reference

/cc @derrabus

Reply:
@jderusse This is `Driver` but `getCurrentTimestampStatement` references `Platforms`.

Reply:
both methods need to do a special thing depending on the database engine. It's weird that one method use the platform and the second uses the driver.

I've no strong opinion about the preferred solution, but I believe both should use the same strategy.

Reply:
Drivers can and will be wrapped with DBAL 3's middleware architecture. Check the platform instead.